### PR TITLE
[FIX] Epic Games tracking on Windows

### PR DESCRIPTION
**Pull request by flavioislima**
This was a fix introduced on Heroic some time ago to fix the playtime for Legendary games.
Should fix the issue with Quest tracking for Epic games on the overlay as well.

It basically uses powershell and the `-Wait` command to wait all the tree processes to be closed before finishing the `callRunner`command.

---

Use the following Checklist if you have changed something on the Backend or Frontend:

- [ ] Tested the feature and it's working on a current and clean install.
- [ ] Tested the main App features and they are still working on a current and clean install. (Login, Install, Play, Uninstall, Move games, etc.)
- [ ] Created / Updated Tests (If necessary)
- [ ] Created / Updated documentation (If necessary)

### CODE REVIEW

**Review by jiyuu-jin — APPROVED** [review]

File: src/backend/launcher.ts [hunk 1]
```diff
@@ -769,20 +770,51 @@ interface RunnerProps {
...
+      .join(',')
+    commandParts = [
+      'Start-Process',
+      `"\`"${fullRunnerPath}\`""`,
+      '-Wait',
+      '-NoNewWindow'
+    ]
+    if (argsAsString) commandParts.push('-ArgumentList', argsAsString)
+
+    bin = fullRunnerPath = 'powershell'
```
Comment (BrettCleary) [inline src/backend/launcher.ts:815]: nitpick but can we break this up into two lines. we don't need the code to be this terse

File: src/backend/launcher.ts [hunk 2]
```diff
@@ -769,20 +770,51 @@ interface RunnerProps {
...
+
+  // macOS/Linux: `spawn`ing an executable in the current working directory
+  // requires a "./"
+  if (!isWindows) bin = './' + bin
+
+  // On Windows: Use PowerShell's `Start-Process` to wait for the process and
+  // its children to exit, provided PowerShell is available
+  if (shouldUsePowerShell === null) {
+    shouldUsePowerShell =
+      isWindows && !!(await searchForExecutableOnPath('powershell'))
```
Comment (BrettCleary) [inline src/backend/launcher.ts:799]: might be more readable if we just assign this to a variable first 
`const powershellPath = await searchForExecutableOnPath('powershell')`

File: src/backend/launcher.ts [hunk 3]
```diff
@@ -983,11 +1013,25 @@ function getRunnerCallWithoutCredentials(
   const modifiedCommand = [...command]
   // Redact sensitive arguments (Authorization Code for Legendary, token for GOGDL)
```
Comment (BrettCleary) [inline src/backend/launcher.ts:1014]: could make this a separate method called `redactSensitiveArguments`

**Review by BrettCleary — APPROVED** [review]
Nice! Let's have @nyghtstalker test the launch user paths for this one

File: src/backend/launcher.ts (hunk 1 above)
Comment (flavioislima) [reply src/backend/launcher.ts:815]: "Modern Javascript" ahahahhahaha

File: src/backend/launcher.ts (hunk 3 above)
Comment (flavioislima) [reply src/backend/launcher.ts:1014]: yeah, I think it is something like this in heroic right now, I will do that in another PR though, there are other changes I want to get for this launcher file from there as well.

**Review by nyghtstalker — APPROVED** [review]
Works great! I am seeing playtime tracked and displayed on the game page. 

image

**Comment by BrettCleary** [comment]
epic play session tracking is working

**Pull request merged**